src/Entity.cpp

```diff
@@ -11,6 +11,14 @@ m_size( j["width"].get<int>(), j["height"].get<int>() ),
 m_position( j["px"][0].get<int>(), j["px"][1].get<int>() ),
 m_grid_pos( j["__grid"][0].get<int>(), j["__grid"][1].get<int>() )
 {
+    if (j["__tile"] != nullptr)
+    {
+        m_tileset = &w->getTileset(j["__tile"]["tilesetUid"].get<int>());
+        m_src_rect.x = j["__tile"]["srcRect"][0];
+        m_src_rect.y = j["__tile"]["srcRect"][1];
+        m_src_rect.width = j["__tile"]["srcRect"][2];
+        m_src_rect.height = j["__tile"]["srcRect"][3];
```

Comment:
I'm just nitpicking here, I like to keep things most explicit possible (and also to stay coherent with  the rest of the project)
```suggestion
    if (!j["__tile"].is_null()) {
        m_tileset = &w->getTileset(j["__tile"]["tilesetUid"].get<int>());
        m_src_rect.x = j["__tile"]["srcRect"][0].get<int>();
        m_src_rect.y = j["__tile"]["srcRect"][1].get<int>();
        m_src_rect.width = j["__tile"]["srcRect"][2].get<int>();
        m_src_rect.height = j["__tile"]["srcRect"][3].get<int>();
```